.github/copilot-instructions.md

```diff
@@ -0,0 +1,177 @@
+# GitHub Copilot Instructions for alfresco-build-tools
+
+This repository contains shared/reusable CI configurations for GitHub Actions serving the Alfresco organization and broader community.
+
+## Primary Development Guidelines
+
+### Version Management (CRITICAL)
+**Always bump the version in `version.txt` for every PR that contains functional changes** (not documentation-only changes):
+
+- **Patch version bump (x.y.Z)**: Bug fixes in existing actions
+- **Minor version bump (x.Y.z)**: New actions or backward-compatible improvements  
+- **Major version bump (X.y.z)**: Breaking changes requiring users to update their workflows
+
+```bash
+# Example: Update version.txt from v9.0.0 to v9.1.0 for a new action
+echo "v9.1.0" > version.txt
+```
+
+### Documentation Requirements
+
+1. **Always update `docs/README.md`** when adding or modifying GitHub Actions:
+   - Add new action to table of contents
+   - Include usage example with proper YAML syntax
+   - Document all input parameters and outputs
+   - Follow existing documentation patterns
+
+2. **Action Documentation Pattern**:
+```markdown
+### action-name
+
+Brief description of what the action does.
+
+```yaml
+      - uses: Alfresco/alfresco-build-tools/.github/actions/action-name@ref
+        with:
+          parameter: value
+```
+
+Additional notes or configuration details.
+```
+
+3. **Run validation script** to ensure all actions are documented:
+```bash
+./check_readme.sh
+```
+
+## Action Development Guidelines
+
+### File Structure
+- Each action must be in `.github/actions/action-name/`
+- Must contain `action.yml` file with proper metadata
+- Include `README.md` for complex actions
+- Follow existing action patterns for consistency
+
+### Security Best Practices
+- Pin action versions to specific SHA commits in examples
+- Avoid exposing secrets in logs
+- Use `secrets.BOT_GITHUB_TOKEN` for authenticated operations
+- Validate inputs and sanitize user-provided data
+
+### Testing Requirements
+- Add tests to `.github/tests/` for new actions when applicable
+- Test actions locally using the `test/local-actions` label
+- Ensure actions work on both `ubuntu-latest` and `ubuntu-24.04-arm` runners
+- Validate backward compatibility for existing actions
```

Comment:
maybe new actions needs to be added to .github/workflows/test.yml if there are no side effects or secrets requirements or external resources